pkg/runcfanotify/runcfanotify.go

```diff
@@ -175,14 +162,26 @@ func NewRuncNotifier(callback RuncNotifyFunc) (*RuncNotifier, error) {
 	}
 	n.runcBinaryNotify = runcBinaryNotify
 
+	runcFound := false
+
 	for _, runcPath := range runcPaths {
 		file := filepath.Join(hostRoot, runcPath)
-		err = runcBinaryNotify.Mark(unix.FAN_MARK_ADD, unix.FAN_OPEN_EXEC_PERM, unix.AT_FDCWD, file)
-		if err == nil {
-			log.Debugf("Checking %q: done", file)
-		} else {
-			log.Debugf("Checking %q: %s", file, err)
+		if _, err := os.Stat(file); errors.Is(err, os.ErrNotExist) {
+			continue
+		}
+		runcFound = true
+
+		if err := runcBinaryNotify.Mark(unix.FAN_MARK_ADD, unix.FAN_OPEN_EXEC_PERM, unix.AT_FDCWD, file); err != nil {
+			runcBinaryNotify.File.Close()
```

Comment:
> Maybe defer this?

In order to use defer I'll have to change the return values to be named and check if err != nil. IMO it's not worth it in this case. 

>  do you want to try all the paths or you want to stop at the first found? In this case, you can break after the below log.Debugf.

We should check all of them. (It could happen that there are two different runc instances installed on the system and we don't want to miss the containers from one of them).